src/Extractor/CustomTransformerExtractor.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\Extractor;
+
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformerInterface;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Arg;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+
+// todo add cache by memoization
+final readonly class CustomTransformerExtractor
+{
+    private Parser $parser;
+
+    public function __construct(?Parser $parser = null)
+    {
+        $this->parser = $parser ?? (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+    }
+
+    /**
+     * Extracts the code of the "transform()" method from a custom transformer,
+     * and wraps it inside a closure, in order to inject it in the generated mappers.
+     *
+     * @param class-string<CustomModelTransformerInterface> $customTransformerClass
+     */
+    public function extract(string $customTransformerClass, Expr $sourceExpression): Expr
+    {
+        $statements = $this->parser->parse(file_get_contents($filename = (new \ReflectionClass($customTransformerClass))->getFileName()));
+
+        $namespaceStatement = self::findUnique(Stmt\Namespace_::class, $statements, $filename);
+        /** @var Stmt\Class_ $classStatement */
+        $classStatement = self::findUnique(Stmt\Class_::class, $namespaceStatement->stmts, $filename);
+
+        $transformMethod = $classStatement->getMethod('transform') ?? throw new \LogicException(
+            "Cannot find method \"transform()\" in custom transformer class \"{$customTransformerClass}\"."
```

Comment:
Shouldn't exists since the interface makes that method mandatory. Is it really required ?

Reply:
this makes type system happy since `ClassLike::getMethod()` returns a nullable, and this is a way to fail fast if anything goes bad... when dealing with AST code, I'd rather be defensive...